plugin.audio.radiofrance/default.py

```diff
@@ -1,163 +1,112 @@
+import os
+import datetime
 import json
+
 import sys
 import requests
 from urllib.parse import parse_qs
-from enum import Enum
+from concurrent.futures import ThreadPoolExecutor
+import itertools
 
-# http://mirrors.kodi.tv/docs/python-docs/
-# http://www.crummy.com/software/BeautifulSoup/bs4/doc/
-from urllib.parse import urlencode, quote_plus
-from ast import literal_eval
 import xbmc
+import xbmcvfs
 import xbmcgui
 import xbmcplugin
 
 from utils import *
+from interface import *
 
 DEFAULT_MANIFESTATION = 0
 RADIOFRANCE_PAGE = "https://www.radiofrance.fr"
 
+CACHE_FILE = os.path.join(xbmcvfs.translatePath('special://temp/'), 'my_cache.json')
```

Comment:
I would use a name that is unique to your add-on, as `my_cache.json` does not really identify that it belongs to your add-on and might cause issues in the future if another add-on uses the same name.

Reply:
I'd recommentd to store all addon-specific data including cache in the addon profile dir:
`Addon().getAddonInfo('profile')`